src/scripts/config/mlz_config_create.sh

```diff
@@ -70,12 +70,85 @@ if [[ -z $(az ad sp list --filter "displayName eq '${mlz_sp_name}'" --query "[].
         --output tsv)
 
     # Get Service Principal AppId
+    # Added the sleep below to accomodate for the transient behavior where the Service Principal creation
+    # is complete but an immediate query for it will fail. The sleep loop will run for 3 minutes and then
+    # the script will exit due to a platform problem
+
+    sp_exists="az ad sp show \
+        --id http://${mlz_sp_name} \
+        --query appId"
+
+    if ! $sp_exists &> /dev/null; then
+
+        sleep_time_in_seconds=10
+        max_wait_in_minutes=3
+        max_wait_in_seconds=180
+        max_retries=$((max_wait_in_seconds/sleep_time_in_seconds))
+
+        echo "Maximum time to wait in seconds = ${max_wait_in_seconds}"
+        echo "Maximum number of retries = ${max_retries}"
+        
+        count=1
+
+        while ! $sp_exists &> /dev/null
+        do
+
+            echo "Waiting for Service Principal appId to complete provisioning (${count}/${max_retries})"
+            echo "Trying again in ${sleep_time_in_seconds} seconds..."
+            sleep "${sleep_time_in_seconds}"
+
+            if [[ ${count} -eq max_retries ]]; then
+                echo "Provisioning the Service Principal appId has exceeded ${max_wait_in_minutes} minutes. Investigate and re-run script."
+                exit 1
+            fi
+
+            count=$((count +1))
+
+        done
+    fi
```

Comment:
Since we're doing these checks twice, it could make things more maintainable and a bit easier to follow by abstracting this into a function. I wrote that up here for what that would look like:

https://github.com/Azure/missionlz/commit/9438ba41b15419487087fe1fe6418afdfa50cc3e?branch=9438ba41b15419487087fe1fe6418afdfa50cc3e&diff=split

Reply:
Updated code to use function that takes 2 arguments